File: postomate.java
import javax.swing.*;
import java.awt.*;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import java.io.IOException;
import java.util.logging.Level;
import java.util.logging.Logger;
import org.apache.http.client.methods.HttpPost;
import org.apache.http.entity.StringEntity;
import org.apache.http.impl.client.CloseableHttpClient;
import org.apache.http.impl.client.HttpClients;
import org.apache.http.util.EntityUtils;
import twitter4j.Status;
import twitter4j.Twitter;
import twitter4j.TwitterException;
import twitter4j.TwitterFactory;
import twitter4j.conf.ConfigurationBuilder;
import com.github.instagram4j.instagram4j.IGClient;
import com.github.instagram4j.instagram4j.requests.media.MediaConfigureToStoryRequest;
import com.github.instagram4j.instagram4j.requests.media.MediaConfigureTimelineRequest;
import com.github.instagram4j.instagram4j.requests.media.MediaConfigureToStoryRequest.MediaConfigureToStoryPayload;
import com.github.instagram4j.instagram4j.requests.media.MediaConfigureTimelineRequest.MediaConfigurePayload;
import com.github.instagram4j.instagram4j.requests.media.MediaUploadRequest;
import com.github.instagram4j.instagram4j.requests.payload.StatusResult;

public class SocialMediaPostingApp extends JFrame {
    private JTextField messageField;
    private JTextField imageField;

    public SocialMediaPostingApp() {
        setTitle("Social Media Posting");
        setSize(400, 200);
        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);

        JPanel panel = new JPanel();
        panel.setLayout(new GridLayout(3, 2));

        JLabel messageLabel = new JLabel("Message:");
        messageField = new JTextField();

        JLabel imageLabel = new JLabel("Image URL:");
        imageField = new JTextField();

        JButton postButton = new JButton("Post");
        postButton.addActionListener(new ActionListener() {
            @Override
            public void actionPerformed(ActionEvent e) {
                String message = messageField.getText();
                String imageURL = imageField.getText();

                // Send post to Facebook
                try {
                    CloseableHttpClient httpClient = HttpClients.createDefault();
                    HttpPost request = new HttpPost("https://graph.facebook.com/me/feed?access_token=YOUR_FACEBOOK_ACCESS_TOKEN");
                    String json = String.format("{\"message\":\"%s\",\"link\":\"%s\"}", message, imageURL);
                    StringEntity entity = new StringEntity(json);
                    request.setEntity(entity);
                    request.setHeader("Content-Type", "application/json");
                    httpClient.execute(request);
                    httpClient.close();
                    JOptionPane.showMessageDialog(null, "Post sent to Facebook successfully!");
                } catch (IOException ex) {
                    Logger.getLogger(SocialMediaPostingApp.class.getName()).log(Level.SEVERE, null, ex);
                    JOptionPane.showMessageDialog(null, "Error sending post to Facebook: " + ex.getMessage());
                }

                // Send post to Twitter
                try {
                    ConfigurationBuilder cb = new ConfigurationBuilder();
                    cb.setDebugEnabled(true)
                            .setOAuthConsumerKey("YOUR_TWITTER_CONSUMER_KEY")
                            .setOAuthConsumerSecret("YOUR_TWITTER_CONSUMER_SECRET")
                            .setOAuthAccessToken("YOUR_TWITTER_ACCESS_TOKEN")
                            .setOAuthAccessTokenSecret("YOUR_TWITTER_ACCESS_TOKEN_SECRET");

                    TwitterFactory tf = new TwitterFactory(cb.build());
                    Twitter twitter = tf.getInstance();
                    Status status = twitter.updateStatus(message + " " + imageURL);
                    JOptionPane.showMessageDialog(null, "Post sent to Twitter successfully!");
                } catch (TwitterException ex) {
                    Logger.getLogger(SocialMediaPostingApp.class.getName()).log(Level.SEVERE, null, ex);
                    JOptionPane.showMessageDialog(null, "Error sending post to Twitter: " + ex.getMessage());
                }

                // Send post to Instagram
                try {
                    IGClient client = IGClient.builder()
                            .username("YOUR_INSTAGRAM_USERNAME")
                            .password("YOUR_INSTAGRAM_PASSWORD")
                            .login();
                    MediaUploadRequest uploadRequest = new MediaUploadRequest(new java.io.File(imageURL));
                    StatusResult uploadResult = client.sendRequest(uploadRequest);

                    MediaConfigurePayload configurePayload = new MediaConfigurePayload();
                    configurePayload.upload_id = uploadResult.getUpload_id();
                    configurePayload.caption = message;
                    MediaConfigureTimelineRequest configureRequest = new MediaConfigureTimelineRequest(configurePayload);
                    client.sendRequest(configureRequest);

                    MediaConfigureToStoryPayload configureToStoryPayload = new MediaConfigureToStoryPayload();
                    configureToStoryPayload.upload_id = uploadResult.getUpload_id();
                    configureToStoryPayload.caption = message;
                    MediaConfigureToStoryRequest configureToStoryRequest = new MediaConfigureToStoryRequest(configureToStoryPayload);
                    client.sendRequest(configureToStoryRequest);

                    JOptionPane.showMessageDialog(null, "Post sent to Instagram successfully!");
                } catch (IOException ex) {
                    Logger.getLogger(SocialMediaPostingApp.class.getName()).log(Level.SEVERE, null, ex);
                    JOptionPane.showMessageDialog(null, "Error sending post to Instagram: " + ex.getMessage());
                }
            }
        });

        panel.add(messageLabel);
        panel.add(messageField);
        panel.add(imageLabel);
        panel.add(imageField);
        panel.add(new JLabel()); // Empty label for spacing
        panel.add(postButton);

        add(panel);
        setVisible(true);
    }

    public static void main(String[] args) {
        SwingUtilities.invokeLater(new Runnable() {
            @Override
            public void run() {
                new SocialMediaPostingApp();
            }
        });
    }
}

File: postomate.py
import tkinter as tk
from tkinter import messagebox
import requests
import tweepy
from instabot import Bot

# Set up API credentials for Facebook, Twitter, and Instagram
FACEBOOK_ACCESS_TOKEN = 'YOUR_FACEBOOK_ACCESS_TOKEN'
TWITTER_CONSUMER_KEY = 'YOUR_TWITTER_CONSUMER_KEY'
TWITTER_CONSUMER_SECRET = 'YOUR_TWITTER_CONSUMER_SECRET'
TWITTER_ACCESS_TOKEN = 'YOUR_TWITTER_ACCESS_TOKEN'
TWITTER_ACCESS_TOKEN_SECRET = 'YOUR_TWITTER_ACCESS_TOKEN_SECRET'
INSTAGRAM_USERNAME = 'YOUR_INSTAGRAM_USERNAME'
INSTAGRAM_PASSWORD = 'YOUR_INSTAGRAM_PASSWORD'

# Create the GUI application
def send_post():
    message = message_entry.get()
    image_url = image_url_entry.get()

    # Send post to Facebook
    try:
        response = requests.post(
            f'https://graph.facebook.com/me/feed?access_token={FACEBOOK_ACCESS_TOKEN}',
            json={'message': message, 'link': image_url}
        )
        response.raise_for_status()
        messagebox.showinfo('Facebook', 'Post sent to Facebook successfully!')
    except requests.exceptions.RequestException as e:
        messagebox.showerror('Facebook', f'Error sending post to Facebook: {str(e)}')

    # Send post to Twitter
    try:
        auth = tweepy.OAuthHandler(TWITTER_CONSUMER_KEY, TWITTER_CONSUMER_SECRET)
        auth.set_access_token(TWITTER_ACCESS_TOKEN, TWITTER_ACCESS_TOKEN_SECRET)
        api = tweepy.API(auth)
        api.update_status(f'{message} {image_url}')
        messagebox.showinfo('Twitter', 'Post sent to Twitter successfully!')
    except tweepy.TweepError as e:
        messagebox.showerror('Twitter', f'Error sending post to Twitter: {str(e)}')

    # Send post to Instagram
    try:
        bot = Bot()
        bot.login(username=INSTAGRAM_USERNAME, password=INSTAGRAM_PASSWORD)
        bot.upload_photo(image_url, caption=message)
        messagebox.showinfo('Instagram', 'Post sent to Instagram successfully!')
    except Exception as e:
        messagebox.showerror('Instagram', f'Error sending post to Instagram: {str(e)}')

# Create the main window
window = tk.Tk()
window.title('Social Media Posting')
window.geometry('400x200')

# Create labels and entry fields
message_label = tk.Label(window, text='Message:')
message_label.pack()
message_entry = tk.Entry(window, width=40)
message_entry.pack()

image_url_label = tk.Label(window, text='Image URL:')
image_url_label.pack()
image_url_entry = tk.Entry(window, width=40)
image_url_entry.pack()

# Create the post button
post_button = tk.Button(window, text='Post', command=send_post)
post_button.pack()

# Start the main loop
window.mainloop()
